### PR TITLE
Fix CDN URL for stable releases

### PR DESCRIPTION
**Pull request by swissspidy**
## Summary

Noticed when testing the 1.2.1 ZIP file.

If running e.g. 1.2.1 stable, the CDN URL should use that version, not `main`.

`main` should only be used for dev releases that contain a commit hash in the version number.

---

See #5723

### CODE REVIEW

**Comment by swissspidy** [comment]
@choumx Any objections to cherry-picking this into 1.2.1?

File: web-stories.php [hunk 1]
```diff
@@ -52,7 +52,7 @@
 
 $cdn_version = 'main';
 
-if ( false !== strpos( WEBSTORIES_VERSION, '+' ) ) {
+if ( false === strpos( WEBSTORIES_VERSION, '+' ) ) {
 	$pieces      = explode( '+', WEBSTORIES_VERSION );
```
Comment (dreamofabear) [inline web-stories.php:56]: If `WEBSTORIES_VERSION` doesn't contain the string "+", how do we split/explode on it?

**Comment by dreamofabear** [comment]
> @choumx Any objections to cherry-picking this into 1.2.1?

What's the impact?

File: web-stories.php (hunk 1 above)
Comment (swissspidy) [reply web-stories.php:56]: Lol 🤦 Let me fix that real quick

Comment (swissspidy) [reply web-stories.php:56]: Ah forgot that PHP simply returns the full string if the `+` isn't found. No problem there whatsoever:

https://3v4l.org/VgoRX

**Comment by swissspidy** [comment]
> > @choumx Any objections to cherry-picking this into 1.2.1?
> 
> What's the impact?

Right now, if you are running the 1.2 release, all images etc. will be loaded from `https://wp.stories.google/static/main/...` instead of `https://wp.stories.google/static/1.2/...`

So if we make changes to `main` tomorrow, e.g. changing all our templates, you'll get broken images.

That's why it's important to load them from the folder corresponding to your version number.

This PR fixes that, so images are loaded from `https://wp.stories.google/static/1.2.1/...` again as expected.

File: web-stories.php (hunk 1 above)
Comment (dreamofabear) [reply web-stories.php:56]: Should we just remove this line then?

Comment (swissspidy) [reply web-stories.php:56]: One step at a time.

The goal of this section is to fall back to `main` if one is using a development build (which has a commit hash in the version number)

Example:
* I submit a PR to add a new template to the gallery.
* I already added the template images to the `main` folder in the `static-site` branch. 
* Now when I enable that PR on the QA site, I need the correct new template images to be loaded

Yes, it's not ideal, but it has worked well for us so far, and alternative options (that allow removing this code), have been explored in #4752 already.

**tl;dr:**

* I would not remove this entire code for 1.2.1 - just hotfix it.
* #4752 needs a solution in the future

Comment (dreamofabear) [reply web-stories.php:56]: I meant removing just the line that calls `explode()`, or am I misunderstanding?

**Review by dreamofabear — APPROVED** [review]

File: web-stories.php (hunk 1 above)
Comment (swissspidy) [reply web-stories.php:56]: Then plugin versions like `1.2.3+abcf123` could not be handled

**Comment by dreamofabear** [comment]
Cherry-pick sounds fine to me.

File: web-stories.php (hunk 1 above)
Comment (dreamofabear) [reply web-stories.php:56]: "1.2.3+abcf123" would not enter the conditional and `$cdn_version` will remain "main".
